app/src/main/java/com/team41/boromi/dbs/BookDB.java

```diff
@@ -0,0 +1,116 @@
+package com.team41.boromi.dbs;
+
+import android.util.Log;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.OnCompleteListener;
+import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
+import com.google.firebase.firestore.CollectionReference;
+import com.google.firebase.firestore.DocumentReference;
+import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FirebaseFirestore;
+import com.google.firebase.firestore.Query;
+import com.google.firebase.firestore.QueryDocumentSnapshot;
+import com.google.firebase.firestore.QuerySnapshot;
+import com.google.gson.Gson;
+import com.team41.boromi.models.Book;
+import com.team41.boromi.models.User;
+
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import static com.team41.boromi.constants.CommonConstants.DB_TIMEOUT;
+
+@Singleton
+public class BookDB {
+
+	private final static String TAG = "BookDB";
+	private final String DB_COLLECTION = "books";
+	private final FirebaseFirestore db;
+	private final CollectionReference booksRef;
+
+	private final Gson gson = new Gson();
+
+	@Inject
+	public BookDB(FirebaseFirestore db) {
+		this.db = db;
+		booksRef = db.collection(DB_COLLECTION);
+	}
+
+	/**
+	 * Attempts to get all books owned by the user
+	 * @param username The username of the authenticated username
+	 * @return A list of all the books owned by the user
+	 */
+	public ArrayList<Book> getUsersOwnedBooks(String username) {
+		final ArrayList<Book> ownedBooks = new ArrayList<>();
+
+		QuerySnapshot res;
+
+		// Gets all books with the owner field equal to thee uuid
+		try {
+			res = Tasks.await(
+					booksRef.whereEqualTo("owner", username).get(),
+					DB_TIMEOUT,

```

Comment:
I kind of curious about this, does this timeout mean the DB only gets 5 seconds to respond with ALL the data. Or is it the DB can only have 5 seconds where there is no response 🤔